include/arch/arc/arch.h

```diff
@@ -87,55 +87,66 @@ extern "C" {
 		1 << (31 - __builtin_clz(x) + 1) : \
 		1 << (31 - __builtin_clz(x)))
 
-
-#if defined(CONFIG_USERSPACE)
-#define ARCH_THREAD_STACK_RESERVED \
-	(STACK_GUARD_SIZE + CONFIG_PRIVILEGED_STACK_SIZE)
-#else
-#define ARCH_THREAD_STACK_RESERVED (STACK_GUARD_SIZE)
-#endif
-
-
-#if  defined(CONFIG_USERSPACE) && CONFIG_ARC_MPU_VER == 2
+#if CONFIG_ARC_MPU_VER == 2
 /* MPUv2 requires
  *  - region size must be power of 2 and >= 2048
  *  - region start must be aligned to its size
  */
-#define Z_ARC_MPUV2_SIZE_ALIGN(size) POW2_CEIL(STACK_SIZE_ALIGN(size))
+#define Z_ARC_MPU_SIZE_ALIGN(size) POW2_CEIL(STACK_SIZE_ALIGN(size))
+#elif CONFIG_ARC_MPU_VER == 3
+/* MPUv3 requires
+ * - region size must be 32 bytes aligned
+ * - region start must be 32 bytes aligned
+ */
+#define Z_ARC_MPU_SIZE_ALIGN(size) STACK_SIZE_ALIGN(size)
+#endif
+
+
+
+#if  defined(CONFIG_USERSPACE)
+
+#if CONFIG_ARC_MPU_VER == 2
+/* MPU stack guard does not works for MPUv2 as it uses GEN_PRIV_STACK */
```

Comment:
Strictly speaking, CONFIG_GEN_PRIV_STACKS could be made to reserve the needed space for the MPU guard in the privilege area. But the 2K minimum region size makes this, on a practical level, fairly useless. I appreciate simply removing support for MPU stack guards for V2, I never felt it was worth maintaining.

Reply:
Agree with Andrew, this comment is a bit confusing. Simply remove or state that MPU stack guards are not supported for V2 due to size overhead.

Reply:
Ok, I will improve the comments. I agree with andrews, no need to have mpu stack guards even no userspace. Becasue it will consume 2K bytes.  ARC STACK CHECKING is a better option.